lib/validation.js

```diff
@@ -66,6 +68,36 @@ function build (context, compile, schemas) {
   }
 }
 
+function generateFluentSchema (schema) {
+  if (schema.params && schema.params[kFluentSchema]) {
+    schema.params = schema.params.valueOf()
+  }
+
+  if (schema.body && schema.body[kFluentSchema]) {
+    schema.body = schema.body.valueOf()
+  }
+
+  if (schema.querystring && schema.querystring[kFluentSchema]) {
+    schema.querystring = schema.querystring.valueOf()
+  }
+
+  if (schema.query && schema.query[kFluentSchema]) {
+    schema.query = schema.query.valueOf()
+  }
+
+  if (schema.headers && schema.headers[kFluentSchema]) {
+    schema.headers = schema.headers.valueOf()
+  }
+
+  if (schema.response) {
+    Object.keys(schema.response).forEach(code => {
+      if (schema.response[code][kFluentSchema]) {
+        schema.response[code] = schema.response[code].valueOf()
+      }
+    })
+  }
```

Comment:
Would you mind extracting a quick utility and avoid the code duplication?

Reply:
Done!